src/renderer/components/groups/NcnnFileInputsGroup.tsx

```diff
@@ -0,0 +1,73 @@
+import log from 'electron-log';
+import { memo } from 'react';
+import { useContext } from 'use-context-selector';
+import { checkFileExists } from '../../../common/util';
+import { GlobalContext } from '../../contexts/GlobalNodeState';
+import { SchemaInput } from '../inputs/SchemaInput';
+import { GroupProps } from './props';
+
+const ifExists = (file: string, then: () => void) => {
+    checkFileExists(file)
+        .then((exists) => {
+            if (exists) {
+                then();
+            }
+        })
+        .catch((reason) => log.error(reason));
+};
+
+export const NcnnFileInputsGroup = memo(
```

Comment:
Groups are going to be super difficult to use if they require whole frontend changes just to make a single group. Is there not a way you can make it dynamically create a group from a backend group definition?

Reply:
No. Groups are kinda like inputs: you have to define a component for each group.

However, groups are reusable. E.g. any number of nodes can use the NCNN group.

Reply:
In that case, then it shouldn't just be an "ncnn" group, it should be a "linked file" group or something a little more generic. 

I still would rather have had something that just lets you combine any number of inputs however you want. Was there a technical limitation preventing that?

Reply:
> In that case, then it shouldn't just be an "ncnn" group, it should be a "linked file" group or something a little more generic.

The logic that the group implements is specific to NCNN and doesn't make much sense in any other context.

I also want to warn of over abstraction here. I'd rather start with groups that fulfill a very specific purpose, and then generalize later as the need arises.

> I still would rather have had something that just lets you combine any number of inputs however you want.

Could you be more specific? I don't know what "however you want" may imply. I.e. grouping non-adjacent inputs together is kinda out of the question because this doesn't mix with the tree structure of React (assuming that we want groups to be able to determine the layout of multiple inputs (I want that, at least)).

Reply:
I mean like being able to define any two inputs and say you want the two to be next to each other, and one to have its value determined by the other -- surely this is something possible without having to make a "two-number-inputs-next-to-each-other-where-the-right-one-is-half-of-the-left-one" group

Reply:
Ultimately you're right though, it's probably better to just get it working right now and worry about abstraction later. But then of course, we risk our future selves wishing we had just abstracted it from the beginning because we're stuck with all these hardcoded groups and we find it annoying to have to make new ones all the time ;)